content/english/java-tictactoe/activity-5.md

```diff
@@ -1,7 +1,111 @@
 ---
 title: "5. (Optional) AI Move"
-description: ""
+description: "Use Minimax algorithm to pick move for the computer"
 date: 2020-07-23T00:00:00Z
 weight: 6
-hidden: true
----
\ No newline at end of file
+difficulty: "Intermediate"
+hidden: false
+---
+{{% notice tip %}}
+#### This is a bonus activity!
+To continue, you should have completed the Tic-Tac-Toe program from the previous exercises! 
+#### Prerequisites (from Java: Basics) 
+- activity-1: <a href="../../java-basics/activity-1" target="_blank">Print Statements and Comments</a>
+- activity-2: <a href="../../java-basics/activity-2" target="_blank">Variables and Types</a>
+- activity-3: <a href="../../java-basics/activity-3" target="_blank">Operators</a>
+- activity-4: <a href="../../java-basics/activity-4" target="_blank">Control Structures</a>
+- activity-5: <a href="../../java-basics/activity-5" target="_blank">Methods</a>
+- activity-6: <a href="../../java-basics/activity-6" target="_blank">Object and Classes</a>
+- activity-7: <a href="../../java-basics/activity-7" target="_blank">Data Structures (Array & ArrayList)</a>
+{{% /notice %}}
+
+### Let's make the computer think
+Previously, we use a `Random` object to randomly generate a move for the computer. As a result, the competitiveness of the computer is fairly low.
+
+In this exercise, we want to increase the difficulty of the game by making optimal decisions for the computer.
+
+We will do so by adding artificial intelligence into our program with the use the <b>Minimax</b> algorithm (a well-defined procedure that allows computers to solve problems).
+
+### Minimax Algorithm
+Minimax is an algorithm that is used in two-player games to make optimal decisions for a player. 
+
+- The algorithm makes optimal decision for the maximizer by examining all the possible future game states based on the current game board. 
+- The two players are labeled as <b>maximizer</b> and <b>minimizer</b> respectedly. In this case, the computer is the maximizer, and the player is the minimizer. 
+- If there is a winner or tie, that game state will be given a score.
+- In our game specifically, we want the computer to win in fewest steps possible. Hence, we design the calculation of score for the game board as the following: 
+
+   (computer wins (`1 * number of spots left + 1`), player wins (`-1 * number of spots left + 1`), tie (`0`)).
```

Comment:
In the first maximizing round, the score in the picture is +1*3 = 3, right?
According to the formula above, shouldn't it be 1*2+1 = 4, Since there are 2 spots left after the move?

Reply:
It's supposed to be 1 * (number of spots left + 1) => which make the score 1 *(2+1) = 1*3 = 3 since there are 2 spots left after the move. I will change the files to have parentheses!